docs/marshal.rst

```diff
@@ -44,6 +44,32 @@ Protocol buffer type                Python type             Nullable
     If you *write* a timestamp field using a Python ``datetime`` value,
     any existing nanosecond precision will be overwritten.
 
+.. note::
+
+   Setting a ``bytes`` field from a string value will first base64 decode the string.
+   This is necessary to preserve the original protobuf semantics when converting between
+   Python dicts and proto messages.
+   Converting a message containing a bytes field to a dict will
+   base64 encode the bytes field.
+
+.. code-block:: python
+
+  import proto
+  from google.protobuf.json_format import ParseDict
+
+  class MyMessage(proto.Message):
+      data = proto.Field(proto.BYTES, number=1)
+
+  msg = MyMessage(data=b"this is a message")
+  msg_dict = MyMessage.to_dict(msg)
+
+  assert msg_dict == {'data': 'dGhpcyBpcyBhIG1lc3NhZ2U='}
```

Comment:
That assertion conveys no value to the user -- better would be to use a constant for the un-encoded string value above, and assert that it is encoded here, e.g.:
```python
  assert msg_dict == {'data': base64.urlsafe_b64encode(message_data})
```